File: src/main/java/com/ecommerce_backend/backend/core/domain/Payment.java
package com.ecommerce_backend.backend.core.domain;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

public record Payment(
        UUID id,
        UUID orderId,
        BigDecimal amount,
        PaymentMethod method,
        PaymentStatus status,
        String transactionId,
        String gatewayResponse,
        LocalDateTime createdAt,
        LocalDateTime processedAt
) {
    public Payment {
        if (createdAt == null) {
            createdAt = LocalDateTime.now();
        }
    }
}

File: ecommerce-frontend/src/domain/entities/Payment.ts
export interface Payment {
  id: string;
  orderId: string;
  amount: number;
  method: PaymentMethod;
  status: PaymentStatus;
  transactionId: string | null;
  gatewayResponse: string | null;
  createdAt: string;
  processedAt: string | null;
}

export type PaymentMethod = 'CREDIT_CARD' | 'DEBIT_CARD' | 'PIX' | 'BOLETO';

export type PaymentStatus = 'PENDING' | 'PROCESSING' | 'COMPLETED' | 'FAILED' | 'CANCELED' | 'REFUNDED' | 'PARTIALLY_REFUNDED';

export interface PaymentRequest {
  orderId: string;
  method: PaymentMethod;
  cardNumber: string;
  cardHolderName: string;
  expiryDate: string;
  cvv: string;
  email: string;
  billingAddress?: string;
  postalCode?: string;
}

export interface RefundRequest {
  amount: number;
}